haystack/document_store/faiss.py

```diff
@@ -174,39 +174,37 @@ def update_embeddings(self, retriever: BaseRetriever, index: Optional[str] = Non
         self.faiss_index.reset()
 
         index = index or self.index
-        documents = self.get_all_documents(index=index)
 
-        if len(documents) == 0:
-            logger.warning("Calling DocumentStore.update_embeddings() on an empty index")
-            return
+        document_count = self.get_document_count(index=index)
+        logger.info(f"Updating embeddings for {document_count} docs...")
+        vector_id = self.faiss_index.ntotal
+        page_number = 0
+        for _ in tqdm(range(0, document_count, self.index_buffer_size)):
+            documents = self.get_all_documents(index=index, page_number=page_number, page_size=self.index_buffer_size)
```

Comment:
If we add `page_number` and `page_size` also as parameters of `update_embeddings` function then we don't need `index_buffer_size`. This will also make `update_embeddings` paginated write function.